### DISCUSSION TITLE
using () for grouping might makes more sense rather than an underscore

### DISCUSSION THREAD

**Discussion by sharmapukar217** [rakkasjs/rakkasjs#79] [Ideas]
Generally an underscore `_` is used for something that is excluded or ignored so using `_` for route grouping might be misleading. Instead pair of parentheses `()` is more meaningful.
eg: `_my-group` vs `(my-group)` 

Also, changing the layout.tsx to _layout.tsx visually separated the pages and layouts files ( just sharing my thoughts )

**Comment by cyco130** [comment]
I agree `()` would look better but parentheses require escaping on bash which can cause all sorts of problems down the line.

About the layouts, I was actually thinking of adding an optional file name, like `admin.layout.tsx`, `auth.layout.tsx` etc. because I don't like having lots of open files with the same name on my IDE, I can't tell which is which. I switched from `login/index.page.tsx` to `login/_login.page.tsx` for the same reason.

**Reply by CanRau** [reply]
Yea the escaping is kinda annoying though autocomplete takes care of that for me and I have to say the routing in SvelteKit is pretty nice, the convention of `+page` & `+layout` isn't bad this way it's easy to co-locate components etc and distinguish "routes" but also SolidStarts `<FileRoutes>` virtual component looks really interesting, this way you can mix and match file based with inline code routes or even completely replace `<FileRoutes>`.

**Comment by cyco130** [comment]
Actually `[]` also requires escaping so scratch that comment :)

**Comment by cyco130** [comment]
This is [now available](https://github.com/rakkasjs/rakkasjs/pull/213).
